extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/internal/util/CronHelper.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright (c) 2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.binding.astro.internal.util;
+
+import static com.google.common.base.Preconditions.*;
+import static org.apache.commons.lang.StringUtils.EMPTY;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.Month;
+import java.time.ZoneId;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.EnumMap;
+import java.util.Map;
+
+/**
+ * Helper class to be required for Astro Jobs to deal with CRON expressions
+ *
+ * @author Amit Kumar Mondal - Initial Contribution
+ */
+public final class CronHelper {
+
+    /** CRON Constants */
+    private static final String ANY = "*";
+    private static final String SPACE = " ";
+    private static final String SECONDS = "*";
+    private static final String EACH = "/";
+    private static final String DAYS_OF_WEEK = "?";
+
+    public static final String DAILY_MIDNIGHT = "0 0 0 * * ? *";
+
+    /**
+     * Returns CRON expression from the provided {@link Date} instance
+     *
+     * @param date
+     *            the {@link Date} instance
+     * @return the CRON expression
+     * @throws NullPointerException
+     *             if {@code date} is null
+     */
+    public static String createCronFromDate(final Date date) {
```

Comment:
Just a guess: What about an additional `createCronFromCalendar` method? The bindong works a lot with the Calendar type, so you can use it as a wrapper for the `getTime()` (e.g. in [class AstroThingHandler line 217](https://github.com/eclipse/smarthome/pull/3503/files#diff-637f70cb0a947f264138d125132a0a99R217) or [interface Job line 67](https://github.com/eclipse/smarthome/pull/3503/files#diff-2392a5f8da7ffd48dd46d304d8b68f29R67)